File: java/src/sorter/Consumer.java
package sorter;

/**
 * This class is utilized to receive records using the Provider class
 */
public class Consumer {

    public final Provider source;
    public Consumer(Provider source) {
        this.source = source;
    }

    // Continues to obtain new records until a null record is returned
    void consume() {
        while(true) {
            Record record = source.next();
            if(record == null) return;
        }
    }
}

File: Consumer.cpp
#include "Consumer.h"


/**
 * Initializes consumer constructor
 * @param givenSource the source to get records from
 */
Consumer::Consumer(Provider* givenSource): source(givenSource)  {}

/**
 * Continues to get records until a null record pointer has been reached
 */
void Consumer::consume() {
    while (true){
        Record* ptr = source->next();
        // if null ptr then end consuming
        if (!ptr){
            return;
        }
    }
}
